### ISSUE TITLE
Unknown hard error when parsing some Steam manifests

### ISSUE THREAD

**Issue by UrbanCMC** [erri120/GameFinder#115]
It looks as if the appmanifests created by Steam sometimes contain the full path of the installed application in the `installdir` field. I've noticed this in 3 separate cases and any one of them will crash with the message "Unknown hard error" and the following stacktrace when not running in debug mode:

```
Process terminated. Assertion failed.                                                                                                                                                                                                                                                       
Path is not sanitized: 'g:/programme/steam/steamapps/common/the binding of isaac'                                                                                                                                                                                                           
   at NexusMods.Paths.Utilities.PathHelpers.DebugAssertIsSanitized(ReadOnlySpan`1 path, IOSInformation os, Boolean isRelative)                                                                                                                                                              
   at NexusMods.Paths.RelativePath..ctor(String path)                                                                                                                                                                                                                                       
   at GameFinder.StoreHandlers.Steam.Services.AppManifestParser.<>c__DisplayClass0_0.<ParseManifestFile>b__2(KVValue x) in D:\Programming\CSharp\Projects\_git\GitHub\GameFinder\src\GameFinder.StoreHandlers.Steam\Services\Parsers\AppManifestParser.cs:line 74                           
   at GameFinder.StoreHandlers.Steam.Services.ParserHelpers.<>c__DisplayClass2_0`1.<ParseChildObjectValue>b__0() in D:\Programming\CSharp\Projects\_git\GitHub\GameFinder\src\GameFinder.StoreHandlers.Steam\Services\Parsers\ParserHelpers.cs:line 73                                      
   at FluentResults.Result.Try[T](Func`1 action, Func`2 catchHandler)                                                                                                                                                                                                                       
   at GameFinder.StoreHandlers.Steam.Services.ParserHelpers.ParseChildObjectValue[T](KVObject childObject, KVObject parentObject, Func`2 parser) in D:\Programming\CSharp\Projects\_git\GitHub\GameFinder\src\GameFinder.StoreHandlers.Steam\Services\Parsers\ParserHelpers.cs:line 72      
   at GameFinder.StoreHandlers.Steam.Services.ParserHelpers.<>c__DisplayClass6_0`1.<ParseRequiredChildObject>b__0(KVObject childObject) in D:\Programming\CSharp\Projects\_git\GitHub\GameFinder\src\GameFinder.StoreHandlers.Steam\Services\Parsers\ParserHelpers.cs:line 130              
   at FluentResults.Result`1.Bind[TNewValue](Func`2 bind)                                                                                                                                                                                                                                   
   at GameFinder.StoreHandlers.Steam.Services.ParserHelpers.ParseRequiredChildObject[T](KVObject parentObject, String childObjectName, Func`2 parser) in D:\Programming\CSharp\Projects\_git\GitHub\GameFinder\src\GameFinder.StoreHandlers.Steam\Services\Parsers\ParserHelpers.cs:line 130
   at GameFinder.StoreHandlers.Steam.Services.AppManifestParser.ParseManifestFile(AbsolutePath manifestPath) in D:\Programming\CSharp\Projects\_git\GitHub\GameFinder\src\GameFinder.StoreHandlers.Steam\Services\Parsers\AppManifestParser.cs:line 74                                      
   at GameFinder.StoreHandlers.Steam.SteamHandler.FindAllGames()+MoveNext() in D:\Programming\CSharp\Projects\_git\GitHub\GameFinder\src\GameFinder.StoreHandlers.Steam\SteamHandler.cs:line 92                                                                                             
   at GameFinder.Example.Program.LogGamesAndErrors[TGame](IEnumerable`1 results, ILogger logger, Action`1 action) in D:\Programming\CSharp\Projects\_git\GitHub\GameFinder\other\GameFinder.Example\Program.cs:line 215
   at GameFinder.Example.Program.RunSteamHandler(IFileSystem fileSystem, IRegistry registry) in D:\Programming\CSharp\Projects\_git\GitHub\GameFinder\other\GameFinder.Example\Program.cs:line 183
   at GameFinder.Example.Program.Run(Options options, ILogger logger) in D:\Programming\CSharp\Projects\_git\GitHub\GameFinder\other\GameFinder.Example\Program.cs:line 92
   at GameFinder.Example.Program.<>c__DisplayClass1_0.<Main>b__0(Options x) in D:\Programming\CSharp\Projects\_git\GitHub\GameFinder\other\GameFinder.Example\Program.cs:line 77
   at CommandLine.ParserResultExtensions.WithParsed[T](ParserResult`1 result, Action`1 action)
   at GameFinder.Example.Program.Main(String[] args) in D:\Programming\CSharp\Projects\_git\GitHub\GameFinder\other\GameFinder.Example\Program.cs:line 75
```

Examples of the full installation path in appmanifest files:
```
"AppState"
{
	"appid"		"113200"
	"Universe"		"1"
	"LauncherPath"		"G:\\Programme\\Steam\\steam.exe"
	"name"		"The Binding of Isaac"
	"StateFlags"		"4"
	"installdir"		"g:\\programme\\steam\\steamapps\\common\\the binding of isaac"
	"LastUpdated"		"1431776164"
	"UpdateResult"		"0"
	"SizeOnDisk"		"54971560"
	"buildid"		"624794"
    [...]
}
```

```
"AppState"
{
	"appid"		"32460"
	"Universe"		"1"
	"LauncherPath"		"G:\\Programme\\Steam\\steam.exe"
	"name"		"Monkey Island 2: Special Edition"
	"StateFlags"		"4"
	"installdir"		"g:\\programme\\steam\\steamapps\\common\\monkey2"
	"LastUpdated"		"1363801581"
	"UpdateResult"		"0"
	"SizeOnDisk"		"1936029427"
	"buildid"		"52245"
    [...]
}
```

This could probably be resolved by reinstalling the offending titles, but it might be a scenario worth supporting just to avoid crashes hard to debug if they don't happen to a developer.

**Comment by Phezzan** [comment]
^ Anyone who has removed a hard drive with steam games on it (or had one die) is fairly likely to hit this.

**Comment by Nutzzz** [comment]
According to https://github.com/wabbajack-tools/wabbajack/issues/2456 this was resolved by #118 .

**Comment by UrbanCMC** [comment]
Looks like it. I'm no longer seeing any errors when parsing manifests containing full paths.
